packages/autocomplete-js/src/createAutocompleteDom.ts

```diff
@@ -40,17 +43,17 @@ export function createAutocompleteDom<TItem extends BaseItem>({
     props: autocomplete.getRootProps({}),
     ...autocompleteScopeApi,
   });
-  const root = createDomElement('div', {
+  const root = createDomElement(environment, 'div', {
```

Comment:
It would be nice to only specify the environment to `createDomElement` once, and to keep its current signature.

Example:

```ts
function getCreateDomElement(environment) {
  return function createDomElement(previousSignature) {
    // get `environment` from the scope
  }
}
```

Reply:
What is the goal of keeping the same signature? If I understand correctly, as `environment` will be used in `createDomElement`, you won't be able to use it without `getCreateDomElement`.

Reply:
The signature [is a convention](https://reactjs.org/docs/react-api.html#createelement) is several similar APIs. The goal is to get rid of the environment as soon as we can. We don't need to provide it every time, we can specify it just at the start and then forget about it.

That's how you would use it:

```ts
const createDomElement = getCreateDomElement(environment)

createDomElement('div', {});
```

Reply:
I didn't know! That's cool :D I've added it in https://github.com/algolia/autocomplete/pull/572/commits/6821c31d6e14c675a0ba1f894588b5a3c9d8246c